src/main/java/mate/academy/dao/impl/CarDaoImplJdbc.java

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't delete car by ID "
+                    + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT cars.id, cars.model, cars.manufacturer_id, "
+                + "manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars_drivers.driver_id = ? "
+                + "AND cars.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseDataFromResultSet(resultSet));
+            }
+            setCars(cars);
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get car drivers by ID "
+                    + driverId, e);
+        }
+
+    }
+
+    private void setCars(List<Car> cars) {
+        for (Car car : cars) {
+            car.setDrivers(getCarDrivers(car.getId()));
+        }
+    }
+
+    private Car parseDataFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(name, country);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(model, manufacturer);
+        car.setId(id);
+        return car;
+    }
+
+    private List<Driver> getCarDrivers(Long id) {
+        String query = "SELECT drivers.name, drivers.license_number, "
+                + "cars_drivers.driver_id FROM cars_drivers "
+                + "JOIN drivers ON cars_drivers.driver_id = drivers.id "
+                + "WHERE cars_drivers.car_id = ?";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
```

Comment:
Open one Connection in another Connection it's bad practice, try to use just one connection, instead of this you can create another PreparedStatement. Write like second argument in method Connection connection

Reply:
Changed. Thanx.